### PR TITLE
in Subject schema removing MgiAlleleId in favor of PIDName

### LINKED ISSUE #621: MGI alleles need to be PIDname
- [x] Add Mouse Genome Informatics to the list of registries
- [x] Make mgi_alleles take a list of PIDnames

### PR DESCRIPTION
**Pull request by Sun-flow**
closes #621 

Converts field `Subject.mgi_allele_ids` from taking `MgiAlleleId` to taking `PIDName`, and adds `MouseGenomeInformatics` to `Registry`

### CODE REVIEW

**Review by dyf — COMMENTED** [review]
@saskiad can approve, but j/w - what would these values be for something like?
https://www.informatics.jax.org/allele/MGI:3819121

`registry_identififer="MGI:3819121"`
and
`name="Ath30"`
?

**Comment by saskiad** [comment]
> @saskiad can approve, but j/w - what would these values be for something like? https://www.informatics.jax.org/allele/MGI:3819121
> 
> `registry_identififer="MGI:3819121"` and `name="Ath30"` ?

we want that to be `name="Ath30"`, `registry="MGI"`, `registry_identifier="3819121"`

File: src/aind_data_schema/core/subject.py [hunk 1]
```diff
@@ -99,7 +92,7 @@ class Subject(AindCoreModel):
         description="Genotype of the animal providing both alleles",
         title="Genotype",
     )
-    mgi_allele_ids: List[MgiAlleleId] = Field(default=[], title="MGI allele ids")
+    mgi_allele_ids: List[PIDName] = Field(default=[], title="MGI allele ids")
```
Comment (saskiad) [inline src/aind_data_schema/core/subject.py:95]: let's call this `alleles` (both field name and title) and add a description that says "Allele names and persitent IDs"

**Review by saskiad — CHANGES_REQUESTED** [review]
one small change, otherwise great!

**Review by saskiad — APPROVED** [review]

**Pull request merged**